Разметка всех страниц

AnnaDolgova377 opened the pull request:
---
:mortar_board: [Разметка всех страниц](https://up.htmlacademy.ru/htmlcss/35/user/1876783/tasks/5)

AnnaDolgova377 committed:
главная страница

AnnaDolgova377 committed:
каталог

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,301 @@
+
+  <body>
+    <header class="page-header">
+      <img class="page-header-logo" src="" alt="Логотип Седоны">
```
картинку нужно обернуть в ссылку, которая ведет на главную страницу

AnnaDolgova377 replied:
исправила, только без href, как ты ниже написал

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,301 @@
+            </a>
+          </li>
+        </ul>
+        <a class="button button-navigation-link" href="#">Хочу сюда</a>
```
второй класс можно сделать - navigation-link , а то получается ты пишешь, что это сразу и кнопка и ссылка)

AnnaDolgova377 replied:
готово

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
 
+  <body>
+    <header class="page-header">
+      <img class="page-header-logo" src="" alt="Логотип Седоны">
```
нужно обернуть в ссылку, без атрибута href

AnnaDolgova377 replied:
готово

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+      </nav>
+    </header>
+    <main class="main-index">
+      <section class="title">
```
можно еще задать класс например `intro`

AnnaDolgova377 replied:
исправила

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+        <p class="advantages-text">
+          Рассмотрим причины, по которым Седона круче, чем Гранд-Каньон!
+        </p>
+        <h2 class="visually-hidden">
```
этот заголовок уже лишний, со скрытыми заголовками нужно по аккуратнее, мы должны помогать пользователю, но не перегружать его информацией)

AnnaDolgova377 replied:
убрала

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+            <p class="advantages-description">
+              Седона — не аттракцион для туристов, там течёт своя жизнь
+            </p>
+            <img class="advantages-img-1" src="" alt="Седона - настоящий городок">
```
-1 в названии класса лишнее, эта и следующая картинка одинаковые(ширина высота), им не нужно разные классы делать

AnnaDolgova377 replied:
убрала

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+            </p>
+            <img class="advantages-img-1" src="" alt="Седона - настоящий городок">
+          </li>
+          <li class="advantages-item">
```
тут начинается вложенный список, вложен он будет в первый элемент `li` главного списка. Эти три элемента нужно вынести в отдельный список, т.к. в списке могут быть только однородные элементы

AnnaDolgova377 replied:
вроде сделала))

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+        </ul>
+      </section>
+      <!-- сервисы -->
+      <section class="services">
```
а это вложенный список во второй элемент `li` главного списка

AnnaDolgova377 replied:
и этот тоже

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+        <p class="search-hotel-description">
+          Укажите предполагаемые даты поездки,<br>и мы покажем вам лучшие предложения гостиниц в Седоне
+        </p>
+        <a class="button button-search-hotel" href="#">ПОИСК ГОСТИНИЦЫ В СЕДОНЕ</a>
```
второй класс лучше сделать search-hotel-link , чтобы сначала шло название блока, а потом самого этого элемента. и в названии указываем `link` т.к. это ссылка. первый класс button т.к. он отвечает не за саму сущность, а за внешний вид.

AnnaDolgova377 replied:
исправила

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+          Только полезная информация и никакого спама, честное бойскаутское!
+        </p>
+        <form class="subscribe-form" action="https://echo.htmlacademy.ru" method="post">
+          <p class="subscribe-form-email">
```
это не `email` а какая-то обертка, враппер

AnnaDolgova377 replied:
так подойдет?

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+          <p class="subscribe-form-email">
+            <label class="visually-hidden" for="subscribe-email">Email</label>
+            <input type="email" name="subscribe-email" id="subscribe-email" placeholder="Ваш e-mail" required>
+            <button class="button button-subscribe" type="submit">Подписаться</button>
```
`subscribe-button`

AnnaDolgova377 replied:
исправила

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+          </p>
+        </form>
+      </section>
+      <section class="footer-social">
```
тут и везде ниже можно удалить слово footer из названия класса

AnnaDolgova377 replied:
убрала

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+        </h2>
+        <ul class="footer-social-list">
+          <li class="footer-social-item">
+            <a class="button button-social" href="https://vk.com/htmlacademy">
```
второй класс будет `social-link`

AnnaDolgova377 replied:
исправила

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+          Контакты
+        </h2>
+        <address class="contacts-address">
+          <a class="footer-contacts-address-phone" href="tel:88128121212">+7 (812) 812-12-12</a>
```
`contacts-link`

AnnaDolgova377 replied:
исправила

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+          <a class="footer-contacts-address-phone" href="tel:88128121212">+7 (812) 812-12-12</a>
+        </address>
+      </section>
+      <section class="footer-htmlacademy">
```
можно `copyright`

AnnaDolgova377 replied:
исправила

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+          Сайт разработчика
+        </h2>
+        <a class="htmlacademy-link" href="https://htmlacademy.ru/">
+          <img class="footer-logo" src="" alt="Логотип HTML Academy">
```
`copyright-logo`

AnnaDolgova377 replied:
готово

GasparColor commented on index.html:
```diff
@@ -1,12 +1,206 @@
+        <h2 class="visually-hidden">
+          Сайт разработчика
+        </h2>
+        <a class="htmlacademy-link" href="https://htmlacademy.ru/">
```
`copyright-link`

AnnaDolgova377 replied:
готово

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,301 @@
+          </li>
+        </ul>
+      </header>
+      <section class="catalog-hotels">
```
такая же разметка может использоваться не только для поиска отелей. но и для других услуг, сувениров и тд, тут не нужно привязываться к отелям в названии класса

AnnaDolgova377 replied:
оставила просто "catalog"

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,301 @@
+          </fieldset>
+          <fieldset class="catalog-filter-group">
+            <legend class="catalog-filter-title">Стоимость, ₽:</legend>
+            <input class="catalog-filter-min-price" type="number" placeholder="от" value min="0" max="9000">
+            <input class="catalog-filter-max-price" type="number" placeholder="до" value min="0" max="9000">
```
одинаковые элементы, им не нужны разные классы

AnnaDolgova377 replied:
изменила

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,301 @@
+            <input class="catalog-filter-max-price" type="number" placeholder="до" value min="0" max="9000">
+            <input class="catalog-filter-range" type="range" min="0" max="9000" step="100">
+          </fieldset>
+          <button class="button" type="submit">Применить</button>
+          <button class="button" type="reset">Сбросить</button>
```
нужно задать классы, которые будут определять что это за сущность. сейчас добавлен только класс для внешнего вида

AnnaDolgova377 replied:
указала (на большее фантазии не хватило)))

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,301 @@
+          <button class="button" type="reset">Сбросить</button>
+        </form>
+
+        <p class="hotel-search">Найдено гостиниц: 38</p>
```
в названии не нужно привязываться к отелям

AnnaDolgova377 replied:
скорректировала

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,301 @@
+          <option value="expensive">Сначала дорогие</option>
+        </select>
+
+        <a class="button" href="#">
+          <span class="visually-hidden">Показать карточки</span>
+        </a>
+        <a class="button" href="#">
+          <span class="visually-hidden">Показать изображение</span>
+        </a>
+        <a class="button" href="#">
+          <span class="visually-hidden">Показать список</span>
+        </a>
```
однородные элементы лучше объединить в список, и задать всем элементам этого списка классы

AnnaDolgova377 replied:
сделала

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,301 @@
+          <span class="visually-hidden">Показать список</span>
+        </a>
+        <!-- карточки гостиниц -->
+        <ul class="hotel-list">
```
`catalog-list`

AnnaDolgova377 replied:
исправила

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,301 @@
+        </a>
+        <!-- карточки гостиниц -->
+        <ul class="hotel-list">
+          <li class="hotel-card">
```
`catalog-item` или `catalog-card`

GasparColor replied:
ну и дальше по аналогии

AnnaDolgova377 replied:
везде исправила

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,301 @@
+              <img class="hotel-card-img" src="" alt="Amara Resort & Spa">
+              <h3 class="hotel-card-title">Amara Resort & Spa</h3>
+            </a>
+            <table class="hotel-card-price">
```
это не таблица

AnnaDolgova377 replied:
сделала как <p>

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,301 @@
+        <!-- карточки гостиниц -->
+        <ul class="hotel-list">
+          <li class="hotel-card">
+            <a class="hotel-card-link" href="#">
```
в отдельные ссылки нужно обернуть картинку и заголовок

AnnaDolgova377 replied:
обернула

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,301 @@
+              </tr>
+            </table>
+            <a class="button hotel-card-button" href="#">подробнее</a>
+            <button class="button" type="submit">в избранное</button>
```
у кнопки неправильный тип, она не отправляет форму

AnnaDolgova377 replied:
исправила

AnnaDolgova377 committed:
корректировка

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,303 @@
+            <a class="catalog-card-link" href="#">
+              <h3 class="catalog-card-title">Amara Resort & Spa</h3>
+            </a>
+            <p class="catalog-card-desk">Гостиница</p>
```
лучше не использовать сокращения. очень важно чтобы код легко читался и был понятен

AnnaDolgova377 replied:
исправила, указала, как type (тип: гостиница, мотель или апартаменты)

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,303 @@
+            <p class="catalog-card-desk">Гостиница</p>
+            <p class="catalog-card-price">От 4000 ₽</p>
+            <a class="button catalog-card-more-link" href="#">подробнее</a>
+            <button class="button button-add-to-favorites" type="button">в избранное</button>
```
тут и у ссылке которая идёт выше ты привязалась в названии класса к содержимому. могу же в будущем текст изменить, например - "добавить в корзину" или "быстрый заказ" или что-угодно) и на каждое изменение придется менять название класса, или если не будешь менять, он будет запутывать

AnnaDolgova377 replied:
изменила, но в голову, кроме как указать, что это нижняя ссылка, больше ничего не пришло)))

GasparColor commented on index.html:
```diff
@@ -1,12 +1,205 @@
+        </p>
+        <ul class="advantages-list">
+          <li class="advantages-item">
+            <ul class="advantage-list">
```
тут немного перемудрила
у тебя есть главный список из двух элементов, это карточки с картинками. Каждая лишка имеет вложенный список

```
<ul>
  <li>
     ...
     <ul></ul>
  </li>
  <li>
     ...
     <ul></ul>
  </li>
</ul>
```

AnnaDolgova377 replied:
вот теперь вроде правильно сделала

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,303 @@
+            <input class="catalog-filter-price" type="number" placeholder="до" value min="0" max="9000">
+            <input class="catalog-filter-range" type="range" min="0" max="9000" step="100">
+          </fieldset>
+          <button class="button button-submit" type="submit">Применить</button>
```
второй класс лучше сделать - catalog-filter-submit-button

AnnaDolgova377 replied:
готово

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,303 @@
+            <input class="catalog-filter-range" type="range" min="0" max="9000" step="100">
+          </fieldset>
+          <button class="button button-submit" type="submit">Применить</button>
+          <button class="button button-reset" type="reset">Сбросить</button>
```
catalog-filter-reset-button

AnnaDolgova377 replied:
исправила

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,303 @@
+          <button class="button button-reset" type="reset">Сбросить</button>
+        </form>
+
+        <p class="search">Найдено гостиниц: 38</p>
```
catalog-search

AnnaDolgova377 replied:
исправила

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,303 @@
+        </form>
+
+        <p class="search">Найдено гостиниц: 38</p>
+        <select class="select-control" aria-label="Сортировка">
```
catalog-sort

AnnaDolgova377 replied:
готово

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,303 @@
+            <p class="catalog-card-price">От 4000 ₽</p>
+            <a class="button catalog-card-more-link" href="#">подробнее</a>
+            <button class="button button-add-to-favorites" type="button">в избранное</button>
+            <p class="rating-star">
```
catalog-card-rating-start

AnnaDolgova377 replied:
исправила

GasparColor commented on catalog.html:
```diff
@@ -0,0 +1,303 @@
+            <p class="rating-star">
+              <span class="visually-hidden">4 звезды</span>
+            </p>
+            <p class="rating">
```
catalog-card-rating

AnnaDolgova377 replied:
исправила

AnnaDolgova377 committed:
исправляет список на главной

AnnaDolgova377 commented:
исправила ошибки. По картинкам пока не смотри, alt-ы буду еще корректировать

AnnaDolgova377 committed:
корректирует рейтинг

повторно направляю, забыла в редакторе кода сохранить последние изменения по рейтингу
